File: 05_arrays/05_18_rearrange_array_elements_by_sign/rearrange_array_elements_by_sign.py
def rearrange_by_sign(arr: list):
    length = len(arr)
    result = [0] * length
    positive = 0
    negative = 1
    
    for i in range(length):
        if arr[i] < 0:
            result[negative] = arr[i]
            negative += 2
        else:
            result[positive] = arr[i]
            positive += 2
            
    return result

arr = [3, 1, -2, -5, 2, -4]
n = len(arr)
print("Values:")
print(" ".join(map(str, arr)))
result = rearrange_by_sign(arr)
print("Rearranged Values:")
print(" ".join(map(str, result)))

File: 05_arrays/05_18_rearrange_array_elements_by_sign/rearrange_array_elements_by_sign.js
const rearrangeBySign = (arr = []) => {
    const length = arr.length;

    const result = new Array(length).fill(0);
    let positive = 0;
    let negative = 1;

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] < 0) {
            result[negative] = arr[i];
            negative += 2;
        } else {
            result[positive] = arr[i];
            positive += 2;
        }
    }

    return result;
}


const arr = [3, 1, -2, -5, 2, -4];
const n = arr.length;
console.log("Values:");
console.log(arr.join(" "));
const result = rearrangeBySign(arr);
console.log("Rearranged Values:");
console.log(result.join(" "));